### PR TITLE
Annotations: Disable "Add annotation" button when annotations are disabled

### LINKED ISSUE #25580: TimeSeries & Graph: Disable add annotation context menu
I want disable annoying "Add annotation" popup
image

**Steps to reproduce**:
click to chart

**Actual Result**:
annoying popup

**Expected Result**
if annotations disabled, popup not showing

**Environment**:
- Grafana version: 7.0.3

### PR DESCRIPTION
**Pull request by ryantxu**
Fixes https://github.com/grafana/grafana/issues/25580

if the builtin annotations layer is disabled, then we should not show the "Add annotations" button in the UI.

Updates the annotation list page to indicate if things are disabled:
image

TODO:
- [x] Disable annotations in panels
- [x] remove the delete button for builtin annotaions... it is added back immediately anyway!
- [x] indicate that annotations are disabled in the UI somehow

### CODE REVIEW

File: public/app/features/dashboard/components/AnnotationSettings/AnnotationSettingsList.tsx [hunk 1]
```diff
@@ -46,11 +70,11 @@ export const AnnotationSettingsList = ({ dashboard, onNew, onEdit }: Props) => {
               <tr key={`${annotation.name}-${idx}`}>
                 {annotation.builtIn ? (
                   <td style={{ width: '90%' }} className="pointer" onClick={() => onEdit(idx)}>
-                    <Icon name="comment-alt" /> &nbsp; <em className="muted">{annotation.name} (Built-in)</em>
+                    {getAnnotationName(annotation)}
```
Comment (ryantxu) [inline public/app/features/dashboard/components/AnnotationSettings/AnnotationSettingsList.tsx:73]: for sure ugly... but not clear how far I should take this refactor/cleanup!

File: public/app/features/dashboard/state/DashboardModel.ts [hunk 2]
```diff
@@ -1114,10 +1114,18 @@ export class DashboardModel implements TimeModel {
   }
 
   canAddAnnotations() {
-    // If RBAC is enabled there are additional conditions to check.
-    const canAdd = !contextSrv.accessControlEnabled() || Boolean(this.meta.annotationsPermissions?.dashboard.canAdd);
+    // When the builtin annotations are disabled, we should not add any in the UI
+    const found = this.annotations.list.find((item) => item.builtIn === 1);
+    if (found?.enable === false) {
```
Comment (leeoniya) [inline public/app/features/dashboard/state/DashboardModel.ts:1119]:
```suggestion
    if (found?.enable === false || !this.canEditDashboard()) {
```

File: public/app/features/dashboard/state/DashboardModel.ts [hunk 3]
```diff
@@ -1114,10 +1114,18 @@ export class DashboardModel implements TimeModel {
...
+      return false;
+    }
 
-    return this.canEditDashboard() && canAdd;
+    if (!this.canEditDashboard()) {
+      return false;
+    }
```
Comment (leeoniya) [inline public/app/features/dashboard/state/DashboardModel.ts:1125]:
```suggestion
```

**Review by leeoniya — APPROVED** [review]
\o/

**Comment by leeoniya** [comment]
maybe kill the Delete button here as well

image

**Comment by tclausing** [comment]
Any interest in changing the positioning so it doesn't overlay the data tooltip when visible, e.g. fixed in corner of viewport or above or below the tooltip?

**Comment by ryantxu** [comment]
> Any interest in changing the positioning so it doesn't overlay the data tooltip when visible, e.g. fixed in corner of viewport or above or below the tooltip?

Yes -- but not in this PR :)

**Comment by leeoniya** [comment]
tracking here: https://github.com/grafana/grafana/issues/57565

**Pull request merged**